ui/src/shared/components/Annotations.tsx

```diff
@@ -1,37 +1,28 @@
 import React, {Component} from 'react'
 import {connect} from 'react-redux'
+import {withRouter} from 'react-router'
 
-import Annotation from 'src/shared/components/Annotation'
+import AnnotationComponent from 'src/shared/components/Annotation'
 import NewAnnotation from 'src/shared/components/NewAnnotation'
 import {SourceContext} from 'src/CheckSources'
 
-import {ADDING, TEMP_ANNOTATION} from 'src/shared/annotations/helpers'
+import {getSelectedAnnotations} from 'src/shared/selectors/annotations'
+import {ADDING} from 'src/shared/annotations/helpers'
 
-import {
-  updateAnnotation,
-  addingAnnotationSuccess,
-  dismissAddingAnnotation,
-  mouseEnterTempAnnotation,
-  mouseLeaveTempAnnotation,
-} from 'src/shared/actions/annotations'
 import {visibleAnnotations} from 'src/shared/annotations/helpers'
 import {ErrorHandling} from 'src/shared/decorators/errors'
 
-import {AnnotationInterface, DygraphClass, Source} from 'src/types'
-import {UpdateAnnotationAction} from 'src/types/actions/annotations'
+import {Annotation, DygraphClass, Source} from 'src/types'
 
 interface Props {
   dWidth: number
   staticLegendHeight: number
-  annotations: AnnotationInterface[]
+  annotations: Annotation[]
   mode: string
   xAxisRange: [number, number]
   dygraph: DygraphClass
   isTempHovering: boolean
-  handleUpdateAnnotation: (
-    annotation: AnnotationInterface
-  ) => UpdateAnnotationAction
-  handleDismissAddingAnnotation: () => void
+  addingAnnotation?: Annotation
```

Comment:
could you clarify this var name? it seems to me like it should be a boolean. `annotationBeingAdded` is clunky but clearer, for example, if that's what this is